addons/website_portal_sale/models/sale_order.py

```diff
@@ -47,6 +59,34 @@ def _notification_recipients(self, message, groups):
 
         return groups
 
+    @api.model_cr_context
+    def _init_column(self, column_name):
+        """ Initialize the value of the given column for existing rows.
+
+            Overridden here because we skip generating unique access tokens
+            for potentially tons of existing sales orders that won't need it.
+            We do, however, generate it for draft or quotation sales orders.
+        """
+        if column_name != 'access_token':
+            super(SaleOrder, self)._init_column(column_name)
+        else:
+            default_compute = self._fields[column_name].default
+
+            query = """
+                SELECT id FROM "%s"
+                 WHERE "%s" is NULL
+                       AND state in %%s
+            """ % (self._table, column_name)
+            self.env.cr.execute(query, (('draft', 'sent'),))
+            order_ids = self.env.cr.fetchall()
+
+            for order_id, in order_ids: # order_ids is a list of tuples e.g.: [(1,), (2,), ...]
+                default_value = default_compute(self)
+
+                query = 'UPDATE "%s" SET "%s"=%%s WHERE id = %s' % (
+                    self._table, column_name, order_id)
```

Comment:
> Are the parameters from cr.execute() going through a safe_eval or so ?

No but query parameters are only for *values*, you can't use them for db identifier (table or column names for instance) you have to inject those by hand.